File: src/main/java/com/m14n/billib/data/html/BBChartChecker.java
package com.m14n.billib.data.html;

import com.m14n.billib.data.BB;
import com.m14n.billib.data.model.BBChart;
import com.m14n.billib.data.model.BBTrack;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.stream.JsonReader;
import com.m14n.ex.Ex;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.text.ParseException;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.List;

public class BBChartChecker {
    private static Date TODAY;

    static {
        try {
            TODAY = BB.CHART_DATE_FORMAT.parse("2016-12-10");
        } catch (ParseException e) {
            e.printStackTrace();
        }
    }

    public static void main(String... args) throws IOException {
        final Gson theGson = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
        final File theOldRoot = new File(BB.DATA_ROOT + File.separator + "country-songs");
        final File theNewRoot = new File(BB.DATA_ROOT + "_new" + File.separator + "country-songs");

        final Calendar theCalendar = Calendar.getInstance();
        theCalendar.setTime(TODAY);

        int theStop = 0;
        while (theStop < 2) {
            final String theFormatDate = BB.CHART_DATE_FORMAT.format(theCalendar.getTime());
            System.out.print(theFormatDate + " ");
            final File theOldFile = new File(theOldRoot, theFormatDate + ".json");
            final File theNewFile = new File(theNewRoot, "country-" + theFormatDate + ".json");
            if (theOldFile.exists() && theNewFile.exists()) {
                final BBChart theOldChart = readOld(theGson, theOldFile, theFormatDate);
                final BBChart theNewChart = readNew(theGson, theNewFile);
                if (theNewChart.equals(theOldChart)) {
                    System.out.println("EQUEALS");
                } else {
                    System.out.println("DIFFERENT!!!");
                    break;
                }
            } else {
                System.out.println("NO FILES!");
                theStop++;
            }
            theCalendar.add(Calendar.DATE, -7);
        }
    }

    private static BBChart readOld(Gson gson, File oldChart, String date) throws IOException {
        final BBChart theChart = new BBChart();
        theChart.setName("Country");
        theChart.setDate(date);
        List<BBTrack> theTracks = new ArrayList<>();
        final JsonReader theReader = new JsonReader(new FileReader(oldChart));
        theReader.beginArray();
        while (theReader.hasNext()) {
            BBTrack theTrack = gson.fromJson(theReader, BBTrack.class);
            theTrack.setCoverUrl(Ex.addHttpIfNeeded(theTrack.getCoverUrl()));
            theTracks.add(theTrack);
        }
        theChart.setTracks(theTracks);
        return theChart;
    }

    private static BBChart readNew(Gson gson, File newChart) throws FileNotFoundException {
        return gson.fromJson(new FileReader(newChart), BBChart.class);
    }
}

File: src/main/java/com/m14n/billib/data/html/BBChartChecker.kt
package com.m14n.billib.data.html

import com.m14n.billib.data.BB
import com.m14n.billib.data.model.BBChart
import com.m14n.billib.data.model.BBTrack
import kotlinx.serialization.builtins.ListSerializer
import kotlinx.serialization.json.Json
import java.io.File
import java.io.FileReader
import java.util.*

fun main() {
    val theOldRoot = File(BB.DATA_ROOT + File.separator + "country-songs")
    val theNewRoot = File(BB.DATA_ROOT + "_new" + File.separator + "country-songs")

    val theCalendar = Calendar.getInstance()
    theCalendar.time = BB.CHART_DATE_FORMAT.parse("2016-12-10")

    var theStop = 0
    while (theStop < 2) {
        val theFormatDate = BB.CHART_DATE_FORMAT.format(theCalendar.time)
        print("$theFormatDate ")
        val theOldFile = File(theOldRoot, "$theFormatDate.json")
        val theNewFile = File(theNewRoot, "country-$theFormatDate.json")
        if (theOldFile.exists() && theNewFile.exists()) {
            val theOldChart = readOld(theOldFile, theFormatDate)
            val theNewChart = readNew(theNewFile)
            if (theNewChart == theOldChart) {
                println("EQUEALS")
            } else {
                println("DIFFERENT!!!")
                break
            }
        } else {
            println("NO FILES!")
            theStop++
        }
        theCalendar.add(Calendar.DATE, -7)
    }
}

private fun readOld(oldChart: File, date: String): BBChart {
    return BBChart(name = "Country", date = date, tracks = FileReader(oldChart).use {
        Json.parse(
            ListSerializer(BBTrack.serializer()),
            oldChart.readText()
        )
    })
}

private fun readNew(newChart: File): BBChart {
    return Json.parse(BBChart.serializer(), newChart.readText())
}
